makeingtables/src/Components/ComponentDid.js

```diff
@@ -0,0 +1,115 @@
+import React from 'react'
+import Posts from './Posts';
+import Comments from './Comments';
+import Users from './Users';
+import axios from 'axios';
+// import ReactDOM from 'react-dom/clt';
+// import { ReactDOM } from 'react';
+import {BrowserRouter,Routes,Route} from 'react-router-dom'
+import { Link } from 'react-router-dom';
+
+export default class ComponentClass extends React.Component {
+    constructor(props) {
+    super(props);
+  
+    // Initializing the state 
+    this.state = {
+        hello:true,
+        renderType:'posts',
+        items: [],
+      };
+  }
+
+  componentDidMount() {
+    console.log("componentDidMountCalled",this.state.renderType);
+    
+    axios.get(`https://jsonplaceholder.typicode.com/${this.state.renderType}`)
+      .then(res => this.setState({
+        items:res.data
+      }));
+  }
+
+  // componentDidUpdate(prevProps,prevState){
+  //   console.log(this.state.items);
+  //   if(prevState.renderType!==this.state.renderType){
+  //   axios.get(`https://jsonplaceholder.typicode.com/${this.state.renderType}`)
+  //       .then(res => this.setState({
+  //       items:res.data
+  //       }));
+  //   }
+  // }
+
+  handleUpdate=(renderType)=>{
+    if(renderType!==this.state.renderType){
+      console.log("Shaktiman")
+    axios.get(`https://jsonplaceholder.typicode.com/${renderType}`)
+          .then(res => this.setState({
+          items:res.data,
+          renderType:renderType
+          }));
+  }
+}
+
+  del = (id)=>{
+    if (window.confirm("Do you really want to delete?")) {
```

Comment:
using window method , can mange this by making a seprate pop up component in react which ask for user input